File: server/src/main/java/com/chanceToMe/MoonGuWanGu/dto/MemberDto.java
package com.chanceToMe.MoonGuWanGu.dto;

import com.chanceToMe.MoonGuWanGu.model.Member;
import java.util.UUID;
import lombok.Getter;

@Getter
public class MemberDto {
    UUID id;

    String email;

    Long lastGachaTimestamp;

    int remainTicket;

    public MemberDto(Member member) {
        this.id = member.getId();
        this.email = member.getEmail();
        this.lastGachaTimestamp = member.getLastGachaTimestamp();
        this.remainTicket = member.getRemainTicket();
    }
}

File: nest-server/src/dto/response/member.dto.ts
import { UUID } from 'crypto';
import { MemberEntity } from 'src/entity';

export class MemberDto {
  id: UUID;

  email: string;

  lastGachaTimestamp: number | null = null;

  remainTicket: number = 0;

  constructor(id: UUID, email: string, lastGachaTimestamp: number, remainTicket: number) {
    this.id = id;
    this.email = email;
    this.lastGachaTimestamp = lastGachaTimestamp;
    this.remainTicket = remainTicket;
  }

  static fromEntity(member: MemberEntity) {
    return new MemberDto(member.id, member.email, member.lastGachaTimestamp, member.remainTicket);
  }
}
